wip/leetcode-2-add-two-numbers.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * https://leetcode.com/problems/add-two-numbers/
+ */
+
+/**
+ * Definition for singly-linked list.
+ * function ListNode(val) {
+ *     this.val = val;
+ *     this.next = null;
+ * }
+ */
+/**
+ * @param {ListNode} l1
+ * @param {ListNode} l2
+ * @return {ListNode}
+ */
+var addTwoNumbers = function(l1, l2) {
+    const fakeHead = {}
+
+    let current = fakeHead
+    let carry = 0
+
+    while (carry > 0 || l1 || l2) {
+        let val = ((l1 && l1.val) || 0) + ((l2 && l2.val) || 0) + carry
+        carry = Math.floor(val / 10)
+        val = val % 10
+
+        if (l1) l1 = l1.next
```

Comment:
Yep, much simpler than helpers I introduced 👍 I overcomplicated.

Reply:
> btw, I'm thinking about using same lists, and not creating a new one. It'd make things complicated but reduced memory usage.

Maybe I'll get back to it later. For now, I'll close. It's been almost 4 months since I opened this PR. Crazy!